### PR TITLE
Migrate Pipeline environment tasks to Node20

### PR DESCRIPTION
**Pull request by qianz2**
**Task name**: 

Tasks with build config **Node20_229_2** :
_"AzureAppServiceManageV0","AzureAppServiceSettingsV1","AzureCLIV1","AzureCLIV2","AzureFileCopyV2","AzureFileCopyV3","AzureFileCopyV4","AzureFileCopyV5","AzureFunctionAppContainerV1","AzureFunctionOnKubernetesV0","AzureFunctionOnKubernetesV1","AzureKeyVaultV1","AzureKeyVaultV2","AzureMysqlDeploymentV1","AzureNLBManagementV1","AzureRmWebAppDeploymentV3","AzureRmWebAppDeploymentV4","AzureWebAppV1", "AzureWebAppContainerV1","ContainerBuildV0","DockerV0","DockerV1","DockerComposeV0","DockerInstallerV0","DotNetCoreCLIV2"_

Tasks with build config **Node20_229_3**:
_"DotNetCoreInstallerV0","DotNetCoreInstallerV1","DownloadBuildArtifactsV0","DownloadFileshareArtifactsV1","DownloadGitHubNpmPackageV1","DownloadGitHubNugetPackageV1","DownloadGitHubReleaseV0","DuffleInstallerV0","FileTransformV1","FileTransformV2","FuncToolsInstallerV0","GitHubReleaseV0","GitHubReleaseV1","GoV0","GoToolV0","HelmDeployV0","HelmInstallerV0","HelmInstallerV1","IISWebAppDeploymentOnMachineGroupV0","JenkinsDownloadArtifactsV1","KubectlInstallerV0","KubernetesV1","KubernetesManifestV0","MysqlDeploymentOnMachineGroupV1","UseDotNetV2"_

**Description**: Migrate 50 tasks owned by pipeline environments team to Node20. 
Splitting them into two task groups: Node20_229_2 and Node20_229_3.
@vsebesta   @rvairavelu  @manolerazvan  @pipeline-environment-team

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by qianz2** [comment]
@manolerazvan @rvairavelu Here's tasks migration PR that your team owns, could you please review and approve it? Thank you

**Comment by DmitriiBobreshev** [comment]
@qianz2 I believe that files inside Tests/task outputs are generated by unit tests and should not be presented in the repo

**Comment by PhilipsonJoseph** [comment]
hi @qianz2 may i know the use case of the file PreJobExecutionAzureFileCopy.ts under Tasks/AzureFileCopyV4 & Tasks/AzureFileCopyV5 and also as @DmitriiBobreshev suggested i hope the files under Tests/task is not required in repo.

**Comment by qianz2** [comment]
> @qianz2 I believe that files inside Tests/task outputs are generated by unit tests and should not be presented in the repo

test outputs are removed. Thanks for catching it.

**Comment by qianz2** [comment]
> hi @qianz2 may i know the use case of the file PreJobExecutionAzureFileCopy.ts under Tasks/AzureFileCopyV4 & Tasks/AzureFileCopyV5 and also as @DmitriiBobreshev suggested i hope the files under Tests/task is not required in repo.

Hi @PhilipsonJoseph I added PreJobExecutionAzureFileCopy.ts from AzureFileCopyV2/AzureFileCopyV3 for testing purpose, looks that it's working w/o this file, so removing it from AzureFileCopyV4/AzureFileCopyV5, thanks!

File: Tasks/AzureFileCopyV5/task.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 5,
         "Minor": 230,
-        "Patch": 0
+        "Patch": 2
```
Comment (merlynomsft) [inline Tasks/AzureFileCopyV5/task.json:17]: Unneeded version bump?  (V4)

File: Tasks/AzureFunctionAppV2/_buildConfigs/Node20/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  "homepage": "https://github.com/Microsoft/vsts-tasks#readme",
+  "dependencies": {
+    "@types/mocha": "^5.2.7",
+    "@types/node": "^20.3.1",
+    "@types/q": "1.0.7",
+    "agent-base": "6.0.2",
+    "azure-devops-node-api": "11.2.0",
+    "azure-pipelines-task-lib": "^4.4.0",
+    "azure-pipelines-tasks-azure-arm-rest": "^3.226.0",
```
Comment (merlynomsft) [inline Tasks/AzureFunctionAppV2/_buildConfigs/Node20/package.json:26]: Suggest we pull AzureFunctions into a seperate PR

**Review by merlynomsft — APPROVED** [review]

File: Tasks/AzureFunctionAppV2/_buildConfigs/Node20/package.json (hunk 2 above)
Comment (qianz2) [reply Tasks/AzureFunctionAppV2/_buildConfigs/Node20/package.json]: Removed, will put them into another PR

File: Tasks/AzureFileCopyV5/task.json (hunk 1 above)
Comment (qianz2) [reply Tasks/AzureFileCopyV5/task.json]: Reverted it back to 5.230.0

**Comment by qianz2** [comment]
> > hi @qianz2 may i know the use case of the file PreJobExecutionAzureFileCopy.ts under Tasks/AzureFileCopyV4 & Tasks/AzureFileCopyV5 and also as @DmitriiBobreshev suggested i hope the files under Tests/task is not required in repo.
> 
> Hi @PhilipsonJoseph I added PreJobExecutionAzureFileCopy.ts from AzureFileCopyV2/AzureFileCopyV3 for testing purpose, looks that it's working w/o this file, so removing it from AzureFileCopyV4/AzureFileCopyV5, thanks!

Actually taking another look into it, this file is required to be node handler execution file we put in task.json and task.loc.json, we will need it to build node20 tasks. 
`"prejobexecution": {
        "Node10": {
          "target": "PreJobExecutionAzureFileCopy.js"
        },
        "Node16": {
          "target": "PreJobExecutionAzureFileCopy.js",
          "argumentFormat": ""
        }
    },`

**Comment by KonstantinTyukalov** [comment]
> > > hi @qianz2 may i know the use case of the file PreJobExecutionAzureFileCopy.ts under Tasks/AzureFileCopyV4 & Tasks/AzureFileCopyV5 and also as @DmitriiBobreshev suggested i hope the files under Tests/task is not required in repo.
> > 
> > 
> > Hi @PhilipsonJoseph I added PreJobExecutionAzureFileCopy.ts from AzureFileCopyV2/AzureFileCopyV3 for testing purpose, looks that it's working w/o this file, so removing it from AzureFileCopyV4/AzureFileCopyV5, thanks!
> 
> Actually taking another look into it, this file is required to be node handler execution file we put in task.json and task.loc.json, we will need it to build node20 tasks. `"prejobexecution": { "Node10": { "target": "PreJobExecutionAzureFileCopy.js" }, "Node16": { "target": "PreJobExecutionAzureFileCopy.js", "argumentFormat": "" } },`

@qianz2 but if I'm not wrong, AzureFileCopyV4 and AzureFileCopyV5 not node tasks but PowerShell, so why we need to use node execution handler is unclear for me

**Comment by qianz2** [comment]
> > > > hi @qianz2 may i know the use case of the file PreJobExecutionAzureFileCopy.ts under Tasks/AzureFileCopyV4 & Tasks/AzureFileCopyV5 and also as @DmitriiBobreshev suggested i hope the files under Tests/task is not required in repo.
> > > 
> > > 
> > > Hi @PhilipsonJoseph I added PreJobExecutionAzureFileCopy.ts from AzureFileCopyV2/AzureFileCopyV3 for testing purpose, looks that it's working w/o this file, so removing it from AzureFileCopyV4/AzureFileCopyV5, thanks!
> > 
> > 
> > Actually taking another look into it, this file is required to be node handler execution file we put in task.json and task.loc.json, we will need it to build node20 tasks. `"prejobexecution": { "Node10": { "target": "PreJobExecutionAzureFileCopy.js" }, "Node16": { "target": "PreJobExecutionAzureFileCopy.js", "argumentFormat": "" } },`
> 
> @qianz2 but if I'm not wrong, AzureFileCopyV4 and AzureFileCopyV5 not node tasks but PowerShell, so why we need to use node execution handler is unclear for me
> 
> UPD. It seems they were in V2 & V3 tasks but has gone after that. Could removing `PrejobExecution` be the real reason for new tasks? If so, I'm not sure it's worth returning them in this PR

Hi @KonstantinTyukalov , AzureFileCopy tasks are written in Powershell, and we use nodejs to build and test it. I've tried to build it first time without `PreJobExecutionAzureFileCopy.js` file in it or without specifying it in task.json/task.loc.json files, and it failed.  Once they were built successfully and migrated tasks files are generated under `_generated `folder, we don't need this `PreJobExecutionAzureFileCopy.js` file in subsequential builds and test runs. So I'm okay to either remove it or keep it.

**Pull request merged**